plugin.rb

```diff
@@ -63,17 +65,28 @@ def show_quick_messages
       if SiteSetting.quick_message_enabled
         if SiteSetting.quick_message_user_preference
           if ActiveModel::Type::Boolean.new.cast(custom_fields['show_quick_messages'])
-            return true
+            return self.quick_messages_access
           else
             return false
           end
         else
-          return true
+          return self.quick_messages_access
         end
       else
         return false
       end
     end
+    def quick_messages_access
```

Comment:
Someone may want to have QM enabled for everyone with a certain badge, but not require the per user preference... To support that case, this block gets duplicated...

```
          if SiteSetting.quick_message_required_badge > 0
            return self.badge_ids.include? (SiteSetting.quick_message_required_badge)
          else
```

Is there a better way that I could do this without repeating myself?

Reply:
@ryanerwin Commit your changes to the PR and I'll take a look

Reply:
@ryanerwin I've simplified the logic on the badge_access branch by making it exclusionary rather than conditional. This works well in this instance as two of the variables are exclusionary cases: enabled and badge access, i.e. if either is not met then the method has to always return false.